ca/server.go

```diff
@@ -624,66 +668,47 @@ func (s *Server) UpdateRootCA(ctx context.Context, cluster *api.Cluster) error {
 		if err != nil {
 			return errors.Wrap(err, "invalid Root CA object in cluster")
 		}
-		externalCARootPool := updatedRootCA.Pool
-		if rCA.RootRotation != nil {
-			// the external CA has to trust the new CA cert
-			externalCARootPool = x509.NewCertPool()
-			externalCARootPool.AppendCertsFromPEM(rCA.CACert)
-			externalCARootPool.AppendCertsFromPEM(rCA.RootRotation.CACert)
-		}
-
-		// Attempt to update our local RootCA with the new parameters
-		if err := s.securityConfig.UpdateRootCA(&updatedRootCA, externalCARootPool); err != nil {
+		if err := s.securityConfig.UpdateRootCA(&updatedRootCA); err != nil {
 			return errors.Wrap(err, "updating Root CA failed")
 		}
 		if err := SaveRootCA(updatedRootCA, s.rootPaths); err != nil {
 			return errors.Wrap(err, "unable to save new root CA certificates")
 		}
+
+		s.externalCAPool = updatedRootCA.Pool
+		if rCA.RootRotation != nil {
+			// the external CA has to trust the new CA cert
+			s.externalCAPool = x509.NewCertPool()
+			s.externalCAPool.AppendCertsFromPEM(rCA.CACert)
+			s.externalCAPool.AppendCertsFromPEM(rCA.RootRotation.CACert)
+		}
+		urls := s.saveAndFilterExternalCAURLS(ctx, externalCACert, rCA.CACert, cluster.Spec.CAConfig.ExternalCAs)
+		// Replace the external CA with the relevant intermediates, URLS, and TLS config
+		s.externalCA = NewExternalCA(updatedRootCA.Intermediates, &tls.Config{
+			Certificates: s.securityConfig.ClientTLSCreds.Config().Certificates,
+			RootCAs:      s.externalCAPool,
+			MinVersion:   tls.VersionTLS12,
+		}, urls...)
+
 		// only update the server cache if we've successfully updated the root CA
-		logger.Debugf("Root CA %s successfully", setOrUpdate)
+		log.G(ctx).Debugf("Root CA %s successfully", setOrUpdate)
 		s.lastSeenClusterRootCA = rCA
 	}
 
-	// we want to update if the external CA changed, or if the root CA changed because the root CA could affect what
-	// certificate for external CAs we want to filter by
-	if rootCAChanged || externalCAChanged {
+	// we want to update only if the external CA URLS have changed, since if the root CA has changed we already
+	// run similar logic
+	if !rootCAChanged && externalCAChanged {
```

Comment:
Could be `else if externalCAChanged {`, but fine with this if it's more readable to be redundant.